File: 백준/Bronze/2979. 트럭 주차/트럭 주차.cc
#include <bits/stdc++.h>
using namespace std;

int arr[101];
int a,b,c;
int s, e;
int main() {
    cin >> a >> b >> c;

    for (int i=0;i<3;i++){
        cin >> s >> e;

        for (int i=s;i<e;i++){
            arr[i]++;
        }
    }

    int ans = 0;

    for (int i : arr){
        if (i == 1) ans += a;
        if (i == 2) ans += b * 2;
        if (i == 3) ans += c * 3;
    }

    cout << ans;
    

    
    

    




    
    
    

    
    return 0;
}

File: 백준/Bronze/2979. 트럭 주차/트럭 주차.java
import javax.print.DocFlavor;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();
        int[] arr = new int[104];
        int ans = 0;

        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());

        for (int i = 0; i < 3; i++) {
            st = new StringTokenizer(br.readLine());
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());

            for (int j = start; j < end; j++) {
                arr[j]++;
            }
        }
        for (int i = 0; i < 101; i++) {
            if (arr[i] == 1) {
                ans += a;
            } else if (arr[i] == 2) {
                ans += 2 * b;
            } else if (arr[i] == 3) {
                ans += 3 * c;
            }
        }
        System.out.println(ans);
    }
}

